### PR TITLE
Dump the block validation cache whenever we perform an import

### PR DESCRIPTION
**Pull request by ribasushi**
This solves (tested on multiple runs) a problem with folks resurrecting long-out-of-sync nodes via
snapshot imports.

The interface switch to Batching is necessary: startup is too long otherwise
( 8 minutes just to clear everything on a relatively old node )

### CODE REVIEW

File: chain/store/store.go [hunk 1]
```diff
@@ -446,6 +449,43 @@ func (cs *ChainStore) takeHeaviestTipSet(ctx context.Context, ts *types.TipSet)
 	return nil
 }
 
+// FlushValidationCache removes all results of block validation from the
+// chain metadata store. Usually the first step after a new chain import.
+func (cs *ChainStore) FlushValidationCache() error {
+	log.Infof("clearing block validation cache...")
+
+	dsWalk, err := cs.ds.Query(query.Query{KeysOnly: true})
```
Comment (magik6k) [inline chain/store/store.go:457]:
```suggestion
	dsWalk, err := cs.ds.Query(query.Query{KeysOnly: true, Prefix: blockValidationCacheKeyPrefix.String()})
```

Then we should be able to drop the prefix check below

Comment (ribasushi) [reply chain/store/store.go:457]: Prefix won't work :( 
Prefix is for `foo/bar`, it doesn't work for `foo:bar` which is how we assemble the validation cache via `.Instance` higher up

Comment (ribasushi) [reply chain/store/store.go]: @magik6k added a comment to clarify why things are the way they are: https://github.com/filecoin-project/lotus/pull/4287/commits/23f5a99117226fba70fea235c421f1d7b6f26745

**Comment by jsign** [comment]
Tested this PR to see if fixes the importing problem, it does!

BTW:
```
2020-10-10T14:54:10.225Z        INFO    chainstore      store/store.go:455      clearing block validation cache...
2020-10-10T14:54:11.216Z        INFO    chainstore      store/store.go:494      224149 block validation entries cleared.
```
So this cache clearing is very fast. Complete importing is 8 min.

Looking forward to see this merged, I'd like to nuke >20 Lotus nodes `datastore/chain` folder with a fresh snapshot. :)

**Review by magik6k — APPROVED** [review]

**Pull request merged**